src/main/java/com/example/BillingSystem/service/impl/ProductServiceImpl.java

```diff
@@ -210,20 +209,20 @@ public Double[] getBill(){
         return bill;
     }
 
-//    @Override
-//    public ProductSummary getBillSummary(){
-//        ProductSummary productsSummary = productRepository.findAll();
-////        List<Product> productList = (List<Product>) productRepository.findAll();
-//
-//        for (Product product:getAllProducts() ) {
-//            List<Product> productsList = productsSummary.getProductsSummary();
-//            productsSummary.setProductsSummary(productsList);
-//        }
-//        productsSummary.setTotalTax(calculateTotalTax());
-//        productsSummary.setTotalPrice(calculateTotalPrice());
-//        productsSummary.setGrossPrice(calculateGrossPrice());
-//
-//        return productsSummary;
-//    }
+    @Override
+    public ProductSummary getBillSummary(){
+        ProductSummary productsSummary = new ProductSummary();
+//        List<Product> productList = (List<Product>) productRepository.findAll();
+
+        for (Product product:getAllProducts() ) {
+            List<Product> productsList = (List<Product>) productRepository.findAll();
+            productsSummary.setProductsSummary(productsList);
```

Comment:
Why are you iterating over the productList here and again fetching the products from the repository? The product is unused here. right?
Also, you can have some null check to the value returned by the repository because the database is an external dependency and it is not necessary that the database should be available always and there is a chance that sometimes,` (List<Product>) productRepository.findAll()` may return an unexpected value.